spc.py

```diff
@@ -0,0 +1,245 @@
+    if not os.path.exists(image_dir):
+        os.makedirs(image_dir)
+
+    print("Starting image conversion...")
+
+    # loop over the images and do the processing
+    images_per_dir = configuration.ImagesPerDir
+
+    if configuration.BayerPattern.lower() == "rg":
+        bayer_conv = cv2.COLOR_BAYER_RG2RGB
+    if configuration.BayerPattern.lower() == "bg":
+        bayer_conv = cv2.COLOR_BAYER_BG2RGB
+
+    print("\n1. Loading {} images.\m".format(str(total_images)))
+    bundle_queue = Queue()
+    for index, image in enumerate(tqdm(image_list)):
+
+        absdir = os.path.join(image_dir, str(images_per_dir * int(index / images_per_dir)).zfill(5))
+
+        filename = os.path.basename(image)
+
+        if not os.path.exists(absdir):
+            os.makedirs(absdir)
+
+        bundle = {'image_path': image,
+                  'image': cvtools.import_image(os.path.dirname(image), filename, bayer_pattern=bayer_conv),
+                  'image_dir': absdir,
+                  'config': configuration}
+        bundle_queue.put(bundle)
+
+    # Get the number of proceess to use based on CPUs
+    n_threads = max(multiprocessing.cpu_count() - 1, 1)
+
+    # Create the set of processes and start them
+    output_queue = Queue()
+    processes = []
+    for i in range(0, n_threads):
+        p = Process(target=process_bundle_list, args=(bundle_queue, output_queue))
+        p.start()
+        processes.append(p)
+
+    # Monitor processing of the images and save processed images to disk as they become available
+    print("2. Processing Images...")
+
+    entry_list = []
+    use_jpeg = configuration.UseJpeg
+    raw_color = configuration.SaveRawColor
+    images_processed = 0
+
+    for counter in tqdm(range(total_images)) or images_processed <= total_images :
+        try:
+            output = output_queue.get()
+            if output:
+                entry_list.append(output['entry'])
+                output_path = os.path.join(output['entry'][DBCONST.IMG_FNAME], output['entry'][DBCONST.IMG_ID])
+                extension = ".jpeg" if use_jpeg else ".png"
+
+                if raw_color:
+                    cv2.imwrite(os.path.join(output_path + "_rawcolor" + extension), output['features']['rawcolor'])
+                cv2.imwrite(os.path.join(output_path + extension), output['features']['image'])
+                cv2.imwrite(os.path.join(output_path + "_binary.png"), output['features']['binary'])
+                images_processed = images_processed+1
+        except:
+            time.sleep(0.05)
+
+    # Terminate the processes in case they are stuck
+    for p in processes:
+        p.terminate()
+
+    print("3. Exporting spreadsheet results...")
+    df = pandas.DataFrame.from_dict(entry_list)
+    df.to_csv(os.path.join(subdir, 'features.tsv'), index=False, sep='\t')
+
+    # Insert into database
+    print("4. Inserting into Database")
+    insert_database(df, db_path="test.db", table_name='date_sampled')
```

Comment:
test.db should also be imported from constants.
Lets use the DEV_ENV to be "test.db" then PROD_ENV to be the actual "plankton.DB"